lsp-bridge-call-hierarchy.el

```diff
@@ -0,0 +1,197 @@
+
+(defvar lsp-bridge-call-hierarchy-name-width 40
+  "Width of the call name column in the call hierarchy buffer.")
+
+(defvar lsp-bridge-call-hierarchy-path-width 60
+  "Width of the path column in the call hierarchy buffer.")
+
+(defvar lsp-bridge-call-hierarchy-call-max-height 30
+  "Maximum height of the call hierarchy buffer.")
+
+(defvar lsp-bridge-call-hierarchy-preview-height 30
+  "Height of the preview window. (set 0 to disable preview)")
+
+(defvar lsp-bridge-call-hierarchy--frame nil)
+
+(defvar lsp-bridge-call-hierarchy--emacs-frame nil)
+
+(defvar lsp-bridge-call-hierarchy--index 0)
+
+(defvar lsp-bridge-call-hierarchy--overlay nil)
+
+(defvar lsp-bridge-call-hierarchy--popup-response nil)
+
+(defvar lsp-bridge-call-hierarchy--temp-buffers nil)
+
+(defun lsp-bridge-incoming-call-hierarchy ()
+  (interactive)
+  (lsp-bridge-call-file-api "call_hierarchy_incoming"
+                            (lsp-bridge--position)))
+
+(defun lsp-bridge-outgoing-call-hierarchy ()
+  (interactive)
+  (lsp-bridge-call-file-api "call_hierarchy_outgoing"
+                            (lsp-bridge--position)))
+
+(defun lsp-bridge-call-hierarchy-posframe-show (buffer)
+  (let* ((posframe-height (+ lsp-bridge-call-hierarchy-preview-height
+                             (min
+                              (length lsp-bridge-call-hierarchy--popup-response)
+                              lsp-bridge-call-hierarchy-call-max-height)))
+         (posframe-width (+ 6 ;; length of spaces and icon
+                            lsp-bridge-call-hierarchy-name-width
+                            lsp-bridge-call-hierarchy-path-width)))
+    (apply #'posframe-show
+           (get-buffer buffer)
+           :poshandler #'posframe-poshandler-frame-center
+           (list
+            :max-height posframe-height
+            :min-height posframe-height
+            :min-width  posframe-width
+            :max-width  posframe-width
+            :border-width 2
+            :border-color "gray"
+            :accept-focus t
+            ))))
+
+(defun lsp-bridge-call-hierarchy-format-call-string (s l)
+  (if (> (length s) l)
+      (format "%s%s" (substring s 0 (- l (length "..."))) "...")
+    (concat s (make-string (- l (length s)) ?\s))))
+
+(defun lsp-bridge-call-hierarchy-format-call (call)
+  (let ((name (plist-get call :name))
+        (path (plist-get call :rel_path))
+        (icon (cdr (assoc (plist-get call :icon) acm-icon-alist))))
+    (format "%s %s %s\n"
+            (if icon
+                (apply #' acm-icon-build icon)
+             " ")
+            (lsp-bridge-call-hierarchy-format-call-string name lsp-bridge-call-hierarchy-name-width)
+            (lsp-bridge-call-hierarchy-format-call-string path lsp-bridge-call-hierarchy-path-width))))
+
+(defun lsp-bridge-call-hierarchy-maybe-preview ()
+  (when (> lsp-bridge-call-hierarchy-preview-height 0)
+    (other-window 1)
+    (lsp-bridge-call-hierarchy-show)
+    (recenter)
```

Comment:
We need add `(setq-local mode-line-format nil)` here, to avoid preview window show mode-line.